### PR TITLE
[SES5] Simplify management of tuned

### PR DESCRIPTION
**Pull request by swiftgist**
Disassociate the specific role names from their purpose.  Considering
that multiple roles can be assigned to a minion, applying a ceph-osd
profile to a gateway or having identical profiles for each role is
unnecessary.

Signed-off-by: Eric Jackson <ejackson@suse.com>
bnc: 1140723
Port: #1782
-----------------

**Checklist:**
- [ ] Added unittests and or functional tests
- [ ] Adapted documentation
- [ ] Referenced issues or internal bugtracker
- [ ] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
what's the status here @swiftgist

**Comment by swiftgist** [comment]
@susebot run teuthology

**Comment by swiftgist** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
this is a legit error @swiftgist 
```
2019-11-15T11:05:02.249 INFO:teuthology.orchestra.run.target192168000081:Running: "sudo bash -c 'DEV_ENV=true timeout 60m deepsea --log-file=/var/log/salt/deepsea.log --log-level=debug salt-run state.orch ceph.stage.3 --simple-output'"
2019-11-15T11:05:02.781 INFO:teuthology.orchestra.run.target192168000081.stdout:Starting stage: ceph.stage.3
2019-11-15T11:05:05.261 INFO:teuthology.orchestra.run.target192168000081.stdout:Parsing stage ceph.stage.3 steps... fail
2019-11-15T11:05:05.261 INFO:teuthology.orchestra.run.target192168000081.stdout:
2019-11-15T11:05:05.262 INFO:teuthology.orchestra.run.target192168000081.stdout:An error occurred while rendering the stage:
2019-11-15T11:05:05.262 INFO:teuthology.orchestra.run.target192168000081.stdout:    ceph.stage.3
2019-11-15T11:05:05.262 INFO:teuthology.orchestra.run.target192168000081.stdout:
2019-11-15T11:05:05.262 INFO:teuthology.orchestra.run.target192168000081.stdout:Error description:
2019-11-15T11:05:05.263 INFO:teuthology.orchestra.run.target192168000081.stdout:  - Rendering SLS 'base:ceph.stage.3.core.default' failed: Jinja syntax error: no filter named 'difference'; line 160
2019-11-15T11:05:05.263 INFO:teuthology.orchestra.run.target192168000081.stdout:
2019-11-15T11:05:05.263 INFO:teuthology.orchestra.run.target192168000081.stdout:    ---
2019-11-15T11:05:05.263 INFO:teuthology.orchestra.run.target192168000081.stdout:    [...]
2019-11-15T11:05:05.263 INFO:teuthology.orchestra.run.target192168000081.stdout:
2019-11-15T11:05:05.264 INFO:teuthology.orchestra.run.target192168000081.stdout:    {% set deepsea_minions = salt['saltutil.runner']('deepsea_minions.matches') %}
2019-11-15T11:05:05.264 INFO:teuthology.orchestra.run.target192168000081.stdout:    {% set mons = salt['saltutil.runner']('select.minions', roles='mon') %}
2019-11-15T11:05:05.264 INFO:teuthology.orchestra.run.target192168000081.stdout:    {% set mgrs = salt['saltutil.runner']('select.minions', roles='mgr') %}
2019-11-15T11:05:05.264 INFO:teuthology.orchestra.run.target192168000081.stdout:
2019-11-15T11:05:05.264 INFO:teuthology.orchestra.run.target192168000081.stdout:    {% set throughput_minions = deepsea_minions | difference(mons) | difference(mgrs) | join(',') %}    <======================
2019-11-15T11:05:05.264 INFO:teuthology.orchestra.run.target192168000081.stdout:
2019-11-15T11:05:05.265 INFO:teuthology.orchestra.run.target192168000081.stdout:
2019-11-15T11:05:05.265 INFO:teuthology.orchestra.run.target192168000081.stdout:    {% if throughput_minions != "" %}
2019-11-15T11:05:05.265 INFO:teuthology.orchestra.run.target192168000081.stdout:    throughput tuned:
2019-11-15T11:05:05.265 INFO:teuthology.orchestra.run.target192168000081.stdout:      salt.state:
2019-11-15T11:05:05.266 INFO:teuthology.orchestra.run.target192168000081.stdout:    [...]
2019-11-15T11:05:05.266 INFO:teuthology.orchestra.run.target192168000081.stdout:    ---
```

**Comment by swiftgist** [comment]
Apparently, difference wasn't added until Salt 2017.07  docs.saltstack.com/en/latest/topics/jinja/index.html#difference

Let me see what other options there are.

**Comment by jschmid1** [comment]
this is blocking us from releasing a new deepsea for 5.5. We should bump up the priority here

**Comment by swiftgist** [comment]
@susebot run teuthology

**Comment by swiftgist** [comment]
@susebot run teuthology

File: srv/modules/modules/difference.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
```
Comment (jschmid1) [inline srv/modules/modules/difference.py]: whats the reason behind having this module in `srv/modules/modules/difference.py` and in `srv/salt/_modules/difference.py ` ?

Comment (swiftgist) [reply srv/modules/modules/difference.py]: https://github.com/SUSE/DeepSea/pull/1783/files#diff-906fd9295174e9bb5a7ee5ea4b8a6f5fR5

**Pull request merged**